docs/rules/max-lines.md

```diff
@@ -0,0 +1,117 @@
+# enforce a maximum file length (max-lines)
+
+Some people consider large files a code smell. Large files tend to do a lot of things and can make following what's going on hard.
+
+## Rule Details
+
+This rule enforces a maximum number of lines per file, in order to aid in maintainability and reduce complexity.
+
+
+## Options
+
+This rule has a number or object option:
+
+* `"max"` (default `100`) enforces a maximum number of lines in a file
```

Comment:
The default should be 300: https://github.com/eslint/eslint/pull/6321/files#diff-662109f5bdfe0d12503c019cf4ae3ed2R52